department-of-veterans-affairs/caseflow
# CASEFLOW-3388: Create Hearing Type Field on Intake Add/Remove Issues Screen

# CASEFLOW-3388: Create Hearing Type Field on Intake Add/Remove Issues Screen

Resolves [CASEFLOW-3388](https://vajira.max.gov/browse/CASEFLOW-3388)

### Description
In the Caseflow Intake 10182 form entry flow, on the Add/Remove Issues page, we added a 'Hearing type' field to the table that appears at the top of the page if certain values are entered in the Review Form. For the field to be rendered the user must select 'Hearing' as the review option **and** select a hearing type from the related dropdown. If rendered, this field will be placed below the row which contains the selected review option/docket type information.

### Acceptance Criteria
- [X] Add the Hearing Type field on the Intake Add/Remove Issues screen
- [X] The field should properly populate with the selection made on the Review screen
- [X] If nothing was selected, the preference is that the entire field does not show

### Testing Plan
Once merged with #17132 :
1. On the Intake Review Form, for `Which review option did the Veteran request?`, select 'Hearing'.
2. Select a hearing type from the hearing type dropdown that will appear (or don't if testing that the field should not appear if no option is selected).
3. Click the 'Continue to next step' button in the bottom-right corner of the Intake Review Form page.
4. If 'Hearing' was the selected review option, and an entry from the hearing type dropdown was selected, you should see a 'Hearing type' field on the Add/Remove Issue page. Otherwise it will be omitted.


### User Facing Changes

BEFORE:
<img width="728" alt="before-issues" src="https://user-images.githubusercontent.com/99351305/158690949-6e231110-782f-4e0d-a292-51f302d6e167.PNG">
AFTER:
<img width="732" alt="after-issues" src="https://user-images.githubusercontent.com/99351305/158690976-a0c8bf50-9103-45a6-9051-4362a78f6440.PNG">


### Code Documentation Updates
